src/bitcoin/electrum/wallet.rs

```diff
@@ -295,10 +295,10 @@ impl BdkWallet {
         })
     }
 
-    /// Find the first block in the local chain whose height is less than or equal to this.
-    pub fn find_block_at_or_before_height(&self, height: u32) -> BlockChainTip {
+    /// Find the highest block in the local chain whose height is below this.
+    pub fn find_block_before_height(&self, height: u32) -> BlockChainTip {
         for cp in self.local_chain.iter_checkpoints() {
-            if cp.height() <= height {
+            if cp.height() < height {
```

Comment:
As such this may trigger the `unreachable!()` below in case the passed `height` is `0`. We need to special case the genesis block or return `None` (and `unwrap_or(genesis_block)` in the caller). I think the latter is cleaner.